src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -74,6 +74,13 @@ const initialDeviceState: DeviceState = {
 }
 export type MetaMaskLikeProvider = providers.Web3Provider & { isTally?: boolean }
 
+// A subset of wallets which have an EIP-1193-like provider
+export type KeyManagerWithProvider =
```

Comment:
Do we want to note that this is an Ethereum/EIP-1193-like provider in the type name?

Reply:
Good point, the rationale to this naming was more that this is a subset of `KeyManager` that has a `provider`, i.e that for this `KeyManager`, a `provider` object exists whereas none exists for the others.

I'm agnostic to this, on the one hand it would be more declarative to add EIP-1193 in the name, on the other hand, that might be overdescriptive, since the only discriminator is that *there is* a `provider`, not *what* is said provider.

Reply:
Checks out, I understand your logic now 👌